python/dolfinx/wrappers/assemble.cpp

```diff
@@ -211,6 +211,48 @@ void declare_assembly_functions(nb::module_& m)
               mat_add, a, std::span(constants.data(), constants.size()),
               py_to_cpp_coeffs(coefficients), bcs);
         }
+        else if (data_bs[0] == 4)
```

Comment:
`dolfinx::la::MatrixCSR` should really support runtime as well as compile-time block sizes. @chrisrichardson?